core/chains/evm/gas/block_history_estimator.go

```diff
@@ -223,7 +225,13 @@ func (b *BlockHistoryEstimator) GetLegacyGas(_ []byte, gasLimit uint32, maxGasPr
 		return nil, 0, errors.New("BlockHistoryEstimator is not started; cannot estimate gas")
 	}
 	if gasPrice == nil {
-		return nil, 0, errors.Errorf("GasPrice unset. Likely cause is that BlockHistoryEstimator failed to make the first estimation on start, or no suitable transactions are available for estimation (size=%d)", b.size)
+		if !b.initialFetch.Load() {
+			return nil, 0, errors.New("BlockHistoryEstimator has not finished the first gas estimation yet, likely because a failure on start")
+		}
+		b.logger.Warnw(fmt.Sprintf("BlockHistoryEstimator couldn't provide an estimation. This is likely because"+
+			"there aren't any valid transactions to estimate from. Using EvmDefaultGasPriceWei as fallback."),
+			"blocks", b.getBlockHistoryNumbers())
```

Comment:
You can drop the `fmt.Sprintf()` now - I actually thought the linter would catch that :thinking: 
Also, consider rewording without the `BlockHistoryEstimator` prefix, since the logger is already `Named("BlockHistoryEstimator")`.
```suggestion
		b.logger.Warnw("Failed to estimate gas price. This is likely because"+
			"there aren't any valid transactions to estimate from. Using EvmDefaultGasPriceWei as fallback.",
			"blocks", b.getBlockHistoryNumbers())
```